include/deal.II/hp/refinement.h

```diff
@@ -254,6 +254,49 @@ namespace hp
       const ComparisonFunction<typename identity<Number>::type>
         &compare_coarsen = std::less_equal<Number>());
 
+    /**
+     * Adapt which finite element to use on a given fraction of cells.
+     *
+     * Out of all cells flagged for a certain type of adaptation, be it
+     * refinement or coarsening, we will determine a fixed number of cells among
+     * this subset that will be flagged for the corresponding p-adaptive
+     * variant.
+     *
+     * For each of both refinement and coarsening subsets, we will determine a
+     * threshold based on the provided parameter @p criteria containing
+     * indicators for every active cell. In the default case for refinement, all
+     * cells with an indicator larger than or equal to the corresponding
+     * threshold will be considered for p-refinement, while for coarsening all
+     * cells with an indicator less than or equal to the matching threshold are
+     * taken into account. However, different compare function objects can be
+     * supplied via the parameters @p compare_refine and @p compare_coarsen to
+     * impose different decision strategies.
+     *
+     * For refinement, the threshold will be associated with the cell that has
+     * the @p p_refine_fraction times Triangulation::n_active_cells() largest
+     * indicator, while it is the cell with the @p p_refine_coarsen times
+     * Triangulation::n_active_cells() lowest indicator for coarsening.
+     *
+     * Each entry of the parameter @p criteria needs to correspond to an active
+     * cell. Parameters @p p_refine_fraction and @p p_coarsen_fraction need to be
+     * in the interval $[0,1]$.
+     *
+     * @note Preceeding calls of Triangulation::prepare_for_coarsening_and_refinement()
```

Comment:
Do you mean "Following calls" or "Later calls"?

Maybe suggest to call that function first. Or does the current function clear refine/coarsen flags on cells where we will use p-adaptivity, and consequently one *has* to call `prepare_c_and_r` again at a later time (maybe through `execute_c_and_r`)?

Reply:
This was meant as a warning that `prepare_c_and_r` will change `refine` and `coarsen` flags which is the decision basis for this function.

So preceeding or earlier calls of `prepare_c_and_r` might change the output of this function. Maybe I should be more specific like "Don't call `prepare_c_and_r` before this function unless you know what you are doing."